src/lib/chips/chip.ts

```diff
@@ -1,17 +1,154 @@
-import { Component, ElementRef, Renderer } from '@angular/core';
+import {
+  // Classes
+  AfterViewInit,
+  Component,
+  ElementRef,
+  EventEmitter,
+  OnDestroy,
+  Renderer,
 
-@Component({
-  selector: 'md-chip, [md-chip]',
+  // Functions
+  forwardRef
+} from '@angular/core';
+
+import {MdFocusable} from '../core/a11y/list-key-manager';
+
+/*
+ * A reusable component config. Utilized by the MdBasicChip
+ * as well as the child class MdChip.
+ */
+export const MD_BASIC_CHIP_COMPONENT_CONFIG: Component = {
+  selector: 'md-basic-chip, [md-basic-chip]',
   template: `<ng-content></ng-content>`,
   host: {
     // Properties
-    'class': 'md-chip',
     'tabindex': '-1',
-    'role': 'option'
-  }
-})
-export class MdChip {
+    'role': 'option',
+
+    // Attributes
+    '[attr.disabled]': 'disabled',
+    '[attr.aria-disabled]': 'isAriaDisabled',
+
+    // Events
+    '(click)': 'click($event)',
+  },
+  inputs: ['disabled'],
+  outputs: ['didfocus', 'destroy']
+};
+
+/**
+ * The base, unstyled chip component extended by the styled MdChip and any custom chips.
+ */
+@Component(MD_BASIC_CHIP_COMPONENT_CONFIG)
+export class MdBasicChip implements MdFocusable, OnDestroy {
+
+  /* Whether or not the chip is disabled. */
+  protected _disabled: boolean = false;
+
+  /**
+   * Emitted when the chip receives focus.
+   * @type {EventEmitter}
+   */
+  public didfocus = new EventEmitter();
+
+  /**
+   * Emitted when the chip is destroyed.
+   * @type {EventEmitter}
+   */
+  public destroy = new EventEmitter();
```

Comment:
No need to include the `public` keyword; we always omit it since it's the default